test/common/router/router_test.cc

```diff
@@ -1565,6 +1567,79 @@ TEST_F(RouterTest, DirectResponse) {
   EXPECT_EQ(1UL, config_.stats_.rq_direct_response_.value());
 }
 
+TEST_F(RouterTest, DirectResponseWithBody) {
```

Comment:
Can we get an integration test for one of these, just to make sure end to end with bodies works smoothly?